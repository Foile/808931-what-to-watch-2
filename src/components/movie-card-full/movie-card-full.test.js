import React from "react";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import MovieCardFull from "./movie-card-full";

it(`MovieCardFull correctly renders after relaunch`, () => {
  const movie = {id: `1`, name: `Macbeth`, posterImage: ``, previewImage: ``, previewVideoLink: ``, starring: [], genre: `Comedy`};
  const tree = renderer
  .create(<BrowserRouter>
    <MovieCardFull
      films = {[movie]}
      match = {{params: {id: `1`}}} /></BrowserRouter>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
