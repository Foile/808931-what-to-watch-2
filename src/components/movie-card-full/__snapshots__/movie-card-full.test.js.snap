// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardFull correctly renders after relaunch 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        <img
          alt="Background Macbeth"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__info"
        >
          <div
            className="movie-card__desc"
          >
            <h2
              className="movie-card__title"
            >
              Macbeth
            </h2>
            <p
              className="movie-card__meta"
            >
              <span
                className="movie-card__genre"
              >
                Comedy
              </span>
              <span
                className="movie-card__year"
              />
            </p>
            <div
              className="movie-card__buttons"
            >
              <button
                className="btn btn--play movie-card__button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="19"
                  viewBox="0 0 19 19"
                  width="19"
                >
                  <use
                    xlinkHref="#play-s"
                  />
                </svg>
                <span>
                  Play
                </span>
              </button>
              <button
                className="btn btn--list movie-card__button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  height="20"
                  viewBox="0 0 19 20"
                  width="19"
                >
                  <use
                    xlinkHref="#add"
                  />
                </svg>
                <span>
                  My list
                </span>
              </button>
              <a
                className="btn movie-card__button"
                href="/films/1/review"
                onClick={[Function]}
              >
                Add review
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Macbeth"
            height="327"
            src=""
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/details"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="/films/1/reviews"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-card__text movie-card__row"
          >
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  className="movie-card__details-value"
                />
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  className="movie-card__details-value"
                />
              </p>
            </div>
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  undefined m
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Comedy
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  className="movie-card__details-value"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height={175}
                poster=""
                src=""
                width={280}
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              Macbeth
            </h3>
          </a>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
